### ISSUE TITLE
[Notifier] [Telegram] Escaping special characters with MarkdownV2

### ISSUE THREAD

**Issue by thomas2411** [symfony/symfony#42697]
**Symfony version(s) affected**: 5.3.6 and below

**Description**
When sending notification to Telegram that contains any special characters from the list  _ * [ ] ( ) ~` > # + - = | { } . ! ,  except dot which is already escaped, throws exception:

``` 
Message: "Handling "Symfony\Component\Notifier\Message\ChatMessage" failed: Unable to post the Telegram message: Bad Request: can't parse entities: Character '-' is reserved and must be escaped with the preceding '\' (code 400)
```
It is for **default** `parse_mode` or `TelegramOptions::PARSE_MODE_MARKDOWN_V2`

**How to reproduce**

Send a notification via telegram with any of the special characters listed above using **default** `parse_mode` or `TelegramOptions::PARSE_MODE_MARKDOWN_V2`

**Possible Solution**
In the documentation of Telegram Bot API https://core.telegram.org/bots/api#markdownv2-style we can see that we need to escape those characters. I have prepared a Regex for this which we can put here https://github.com/symfony/telegram-notifier/blob/5.3/TelegramTransport.php#L85 , replace `str_replace` with `preg_replace` that will escape those characters with `\`.

If you think this is a good solution please let me know, I will Pull a request.

**Comment by OskarStark** [comment]
Thanks for pointing this out, are you open for a PR and add the regex? It would be great if you can add some tests.

As this is a bugfix it should be applied against 5.3 branch, thank you.

Feel free to ping me in the PR to get a review

**Comment by thomas2411** [comment]
Hello Oskar. Yes, I will prepare a PullRequest. I will try to add a test too.

**Comment by OskarStark** [comment]
Closed in #42721

**Comment by tft7000** [comment]
In my understanding, this code change replaces all markdown characters when I choose `->parseMode(TelegramOptions::PARSE_MODE_MARKDOWN_V2)`. 
Doesn't that defeat the purpose of markdown mode?

As far as I understand, currently my only option is, to use the legacy markdown option `TelegramOptions::PARSE_MODE_MARKDOWN` to send some markdown notification.

Maybe the escaping should only happen, if `TelegramOptions::PARSE_MODE_MARKDOWN_V2` is not explicitly chosen in [TelegramTransport::doSend](https://github.com/symfony/symfony/blob/4530e95e2365f517df04fa26568f370071bc0751/src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php#L75).
